tests/web3/eth_estimateGas2.expected.json

```diff
@@ -12,6 +12,6 @@
   {
     "jsonrpc": "2.0",
     "id": 3,
-    "result": "0x5659"
+    "result": "0xd8cb"
```

Comment:
Can you post here what ganache gets on these examples so we have an idea of the difference?

Reply:
I can't get ganache's output for `eth_estimateGas2.in.json` because it's using a very specific `firefly_addAccount` call. The output ganache gives for the estimateGas call in `eth_estimateGas3.in.json` is `0x5208`, as opposed to firefly's `0xd8cb`.

Reply:
0x5208 is way lower than 0xd8cb. Why the huge discrepancy?

Reply:
Because the binary search is being cut short by the first allotment of gas that results in a successful execution of the transaction, which is the amount that gets returned. Allowing the binary search to finish completely wasn't being reliable and was sometimes returning gas amounts that didn't allow the transaction to complete, presumably because there is some sort of error in the algorithm I wrote.